pilot/platform/cloudfoundry/controller_test.go

```diff
@@ -15,136 +15,164 @@
+			"process-guid-b.cfapps.internal": {
+				Backends: []*api.Backend{
+					{
+						Address: "10.0.50.4",
+						Port:    61009,
+					},
+					{
+						Address: "10.0.60.2",
+						Port:    61001,
 					},
 				},
 			},
-		}
-		client.RoutesOutput.Ret1 <- nil
-		controller = &cloudfoundry.Controller{
-			Client: client,
-			Ticker: ticker,
-		}
-	})
-
-	It("calls all the instance handlers when the cache is invalidated", func() {
-		ih1, ih2 := new(fakeInstanceHandler), new(fakeInstanceHandler)
-		sh1, sh2 := new(fakeServiceHandler), new(fakeServiceHandler)
-
-		controller.AppendInstanceHandler(ih1.Do)
-		controller.AppendInstanceHandler(ih2.Do)
-		controller.AppendServiceHandler(sh1.Do)
-		controller.AppendServiceHandler(sh2.Do)
-
-		stop := make(chan struct{})
-		defer close(stop)
-		go controller.Run(stop)
-
-		By("checking no handlers are called before the ticker fires")
-		Consistently(client.RoutesCalled, "100ms").ShouldNot(Receive())
-		Consistently(ih1.callCount, "100ms").Should(Equal(0))
-		Consistently(ih2.callCount, "100ms").Should(Equal(0))
-		Consistently(sh1.callCount, "100ms").Should(Equal(0))
-		Consistently(sh2.callCount, "100ms").Should(Equal(0))
-
-		By("checking that all handlers are called after the first ticker fires")
-		ticker <- time.Time{}
-		Eventually(client.RoutesCalled).Should(Receive())
-		Eventually(ih1.callCount).Should(Equal(1))
-		Eventually(ih2.callCount).Should(Equal(1))
-		Eventually(sh1.callCount).Should(Equal(1))
-		Eventually(sh2.callCount).Should(Equal(1))
-
-		By("checking that no handlers are called if the cached data is still valid")
-		ticker <- time.Time{}
-		Eventually(client.RoutesCalled).Should(Receive())
-		Consistently(ih1.callCount, "100ms").Should(Equal(1))
-		Consistently(ih2.callCount, "100ms").Should(Equal(1))
-		Consistently(sh1.callCount, "100ms").Should(Equal(1))
-		Consistently(sh2.callCount, "100ms").Should(Equal(1))
-
-		By("checking that all handlers are called again when the cache is invalidated")
-		client.RoutesOutput.Ret0 <- &api.RoutesResponse{
-			Backends: map[string]*api.BackendSet{
-				"other-process-guid-a.cfapps.internal": {
-					Backends: []*api.Backend{
-						{
-							Address: "10.10.2.6",
-							Port:    61006,
-						},
-						{
-							Address: "10.0.41.3",
-							Port:    61009,
-						},
+		},
+	}
+	client.RoutesOutput.Ret1 <- nil
+
+	// initialize object under test
+	controller := &cloudfoundry.Controller{
+		Client: client,
+		Ticker: ticker,
+	}
+
+	ih1, ih2 := new(fakeInstanceHandler), new(fakeInstanceHandler)
+	sh1, sh2 := new(fakeServiceHandler), new(fakeServiceHandler)
+
+	controller.AppendInstanceHandler(ih1.Do)
+	controller.AppendInstanceHandler(ih2.Do)
+	controller.AppendServiceHandler(sh1.Do)
+	controller.AppendServiceHandler(sh2.Do)
+
+	stop := make(chan struct{})
+	defer close(stop)
+	go controller.Run(stop)
+
+	// checking no handlers are called before the ticker fires
+	g.Consistently(client.RoutesCalled, "100ms").ShouldNot(gomega.Receive())
```

Comment:
Is `gomega` ok to use then?
I am fine with both, but it looks like others did not want assertion libraries either.

Reply:
Intention was to make incremental progress.

As mentioned up top, if we want to remove Gomega, we could do this in a follow-up PR.

Reply:
Incremental is fine by me.